pkg/secrethub/acl.go

```diff
@@ -218,50 +228,113 @@ func (s accessRuleService) create(path api.BlindNamePath, permission api.Permiss
 		Permission: permission,
 	}
 
-	if currentAccessLevel.Permission < api.PermissionRead {
-		encryptedTree, err := s.client.httpClient.GetTree(blindName, -1, true)
-		if err != nil {
-			return nil, errio.Error(err)
-		}
+	encrypter := newReencrypter(account, s.client)
 
-		accountKey, err := s.client.getAccountKey()
+	tries := 0
+	for {
+		if currentAccessLevel.Permission < api.PermissionRead {
+			err = encrypter.Add(blindName)
+			if err != nil {
+				return nil, err
+			}
+
+			in.EncryptedDirs = encrypter.Dirs()
+			in.EncryptedSecrets = encrypter.Secrets()
+		}
+		err = in.Validate()
 		if err != nil {
-			return nil, errio.Error(err)
+			return nil, err
 		}
 
-		dirs, secrets, err := encryptedTree.DecryptContents(accountKey)
-		if err != nil {
-			return nil, errio.Error(err)
+		accessRule, err := s.client.httpClient.CreateAccessRule(blindName, accountName, in)
+		if err != api.ErrNotEncryptedForAccounts {
+			return accessRule, err
 		}
+		if tries >= missingMemberRetries {
+			return nil, fmt.Errorf("cannot create access rule: resources controlled by the access rule are simultaneously being created; you may try again")
+		}
+		tries++
```

Comment:
What if `err == nil`?

Reply:
This is another example, just like the other PR reviewed yesterday, where the negated comparison for a specific error also includes the nil case for the error:

```go
if err != api.ErrNotEncryptedForAccounts {
	return accessRule, err
}
```

it's exactly equal to:
```go
if err == nil {
	return accessRule, nil
}
if err != api.ErrNotEncryptedForAccounts {
	return nil, err
}
```

Like before, I'll change to the latter implementation, as after the previous discussion we agreed to value the explicitness (and the readability that comes with that) of the latter over the conciseness and the negligible performance gain (of having just a single error check instead of two) of the former.